src/ops/delay.c

```diff
@@ -69,30 +72,23 @@ static void op_DEL_CLR_get(const void *NOTUSED(data), scene_state_t *ss,
 static void mod_DEL_X_func(scene_state_t *ss, exec_state_t *es,
                            command_state_t *cs,
                            const tele_command_t *post_command) {
-    int16_t i = 0;
     int16_t num_delays = cs_pop(cs);
     int16_t delay_time = cs_pop(cs);
     int16_t delay_time_next;
 
-    if (delay_time < 1) delay_time = 1;  // minimum delay time = 1ms
-
-    delay_time_next = delay_time;  // set first delay time to delay time
-
-    // 0 is the magic number for an empty slot.
-    // Be careful not to set delay.time[i] to 0 before calling this function.
-    while (ss->delay.time[i] != 0 && i != DELAY_SIZE) i++;
+    if (delay_time < 1) delay_time = 1;
 
-    while (i < DELAY_SIZE && num_delays > 0) {
-        // queue the delay
-        delay_common_add(ss, es, i, delay_time_next, post_command);
+    // set first delay time to delay time
+    delay_time_next = delay_time;  
 
+    while ( num_delays > 0 && delay_common_add(ss, es, delay_time_next, post_command) ) 
```

Comment:
the function call is part of the while check instead of using a separate bool variable.  pretty sure this is fine, but I may be missing something.